### ISSUE TITLE
About motion clustering

### ISSUE THREAD

**Issue by anl13** [OpenMonkeyStudio/OMS_Data#2]
Hello! Thanks for your excellent work OpenMonkeyStudio. 
After reading the paper, I have a question: 

<b>How did you determine the cluster labels in Fig.6A?  </b>

I know such 2D visualization spectrum is computed by UMAP. However, UMAP could not directly give you the information of per-point cluster label, right? And I didn't find a description on how to obtain the cluster label here. So I am curious about that, looking forward to your answer! 

Thank you very much!

**Comment by thejanzimmermann** [comment]
Hi An, 

thank you. Do you mean the semantic aspect of it ?

**Comment by anl13** [comment]
> Hi An,
> 
> thank you. Do you mean the semantic aspect of it ?

Yes, the clustering result in the paper.

**Comment by christincha** [comment]
Hello I have the same question. How to get semantic labels of action sequences. Thanks!

**Comment by thejanzimmermann** [comment]
Hi there, sorry for the delay. You need an ethogram of behavior first and then you can compute the probability distribution from the cluster centers.

________________________________

**Comment by thejanzimmermann** [comment]
I think it’s also important to note that you will need large datasets to perform meaningful classifications.

________________________________

**Comment by anl13** [comment]
Do you mean we should follow the procedure below: <b>(1)</b> we collect a large amount of data <b>(2)</b> we have clustered all data into some clusters using unsupervised method (like UMAP) ;  <b>(3)</b> manually assign semantic label to each cluster by inspecting the cluster center;  <b>(4)</b> for a new data, we compute the distance to cluster center to determine its own label?

**Comment by thejanzimmermann** [comment]
Exactly, and always best to do out of sample validation :)

**Comment by christincha** [comment]
Thanks for the reply.
I was wondering if there any ground truth label? That is labeled by some annotators.
> Exactly, and always best to do out of sample validation :)

**Comment by anl13** [comment]
Thanks a lot! I still have a question. 

What does "sample validation" mean? 
> Exactly, and always best to do out of sample validation :)

**Comment by christincha** [comment]
I still have another question. In the paper, you mentioned 3D representations are those transformed coordinated, and the y-axis is the gravity direction. Is there any information in the dataset that can provide what is the gravity direction? Or it can be some direction vertical to the z-axis.

**Comment by anl13** [comment]
I assume that, the coordinate system is:  y-axis points towards the ground, which is `gravity direction`

> I still have another question. In the paper, you mentioned 3D representations are those transformed coordinated, and the y-axis is the gravity direction. Is there any information in the dataset that can provide what is the gravity direction? Or it can be some direction vertical to the z-axis.

**Comment by thejanzimmermann** [comment]
The values here are not normalized so be careful. The data provided here is purely for model generation and inference. The actual experimental data is continuous for many hours and we re-reference the cameras intrinsic and extrinsic properties to the cages true coordinate system.

**Comment by christincha** [comment]
Thanks for replying. So is the data provided here is already in the cage coordinate system? Could you please share the clustering result?

Thanks!:)

**Comment by thejanzimmermann** [comment]
No the data is in an arbitrary reference frame that is related to one of the camera. I can check which one.  We are not sharing behavioral data apart from the training data at this point as we are reading manuscripts on those datasets at the moment. Once those manuscripts are on a preprint server we are happy to share those full datasets as well but you will have to provide physical hard drives. One hour of data is 6 terabytes of image data.

________________________________

**Comment by christincha** [comment]
Thanks a lot for replying. Looking forward for your reply. It would be great if you can share how to use intrinsic and extrinsic camera information to register the frame.

**Comment by kilometer0101** [comment]
Hello. May I ask you to publish the annotation labels for the 6 pose clusters (jumping, sitting, ...) for each frame of these batch files?

**Comment by thejanzimmermann** [comment]
Hey there, the data in this package is not the sequence used for the clustering but the raw data for model training. For the sequence used in the paper, the raw data would be multiple terabytes.
________________________________

**Comment by kilometer0101** [comment]
Thanks for the reply. Would it be possible to assign posture labels for arbitrary motion data if you have a trained detector?

**Comment by thejanzimmermann** [comment]
Sure, all you need is someone semi trained on primatology but rough labels such as walking sitting etc can be easily assigned
________________________________

**Comment by kilometer0101** [comment]
In other words, by using the separator you built-in Fig.6 of your paper, is it possible to assign frame by frame labels to the posture data in the published batch files?

**Comment by thejanzimmermann** [comment]
The published data is raw training data (single frames / lots of them) without temporal sequence to them. We did not publish sequence data with this manuscript as the size is too large. A typical dataset from our system for a daily acquisition is 12+ terabyte.
________________________________

**Comment by kilometer0101** [comment]
OK. Thanks a lot!

**Comment by kilometer0101** [comment]
Hello, again. May I ask you the frame rate of this data? 30 fps?

**Comment by thejanzimmermann** [comment]
Yes exactly
